### PR TITLE
Serialize grids with CPU architecture

### LINKED ISSUE #2179: Serialize grid with `CPU` architecture?
I think it might make sense to serialize grids in JLD2 files with `CPU` architecture by default; this way they can be loaded on systems without a GPU. 

The downside is that users with GPUs would have to write

```julia
cpu_grid = file["serialized/grid"]
grid = on_architecture(GPU(), cpu_grid)
```

to load their grid on the GPU.

### PR DESCRIPTION
**Pull request by glwagner**
Closes #2179

### CODE REVIEW

**Comment by navidcy** [comment]
We get a bunch of warnings from:
https://github.com/CliMA/Oceananigans.jl/blob/484f9a2868023918b78e494f34671849d2b44ea5/src/OutputWriters/jld2_output_writer.jl#L209

File: src/OutputWriters/output_writer_utils.jl [hunk 1]
```diff
@@ -27,7 +28,7 @@ saveproperty!(file, location, p::Tuple) =
 saveproperty!(file, location, p) =
     [saveproperty!(file, location * "/$subp", getproperty(p, subp)) for subp in propertynames(p)]
 
-saveproperty!(file, location, p::ImmersedBoundaryGrid) = saveproperty!(file, location, p.grid)
+saveproperty!(file, location, p::ImmersedBoundaryGrid) = saveproperty!(file, location, grid)
```
Comment (navidcy) [inline src/OutputWriters/output_writer_utils.jl:31]: I don't understand where from the `saveproperty!` knows `grid`.

File: src/OutputWriters/output_writer_utils.jl [hunk 2]
```diff
@@ -53,10 +54,14 @@ serializeproperty!(file, location, p::AbstractArray) = saveproperty!(file, locat
 serializeproperty!(file, location, p::Function) = @warn "Cannot serialize Function property into $location"
 serializeproperty!(file, location, p::ContinuousBoundaryFunction) = @warn "Cannot serialize ContinuousBoundaryFunction property into $location"
 
+# Serializing grids:
+serializeproperty!(file, location, p::AbstractGrid) = file[location] = on_architecture(CPU(), grid)
```
Comment (navidcy) [inline src/OutputWriters/output_writer_utils.jl:58]: same here, I don't see where from `serializeproperty!` knows about `grid`

File: src/OutputWriters/output_writer_utils.jl (hunk 1 above)
Comment (glwagner) [reply src/OutputWriters/output_writer_utils.jl]: Thats a typo / bug.

File: src/OutputWriters/output_writer_utils.jl (hunk 2 above)
Comment (glwagner) [reply src/OutputWriters/output_writer_utils.jl]: Bug!

Comment (navidcy) [reply src/OutputWriters/output_writer_utils.jl]: 🐞🐛🐜

**Review by navidcy — APPROVED** [review]

**Pull request merged**